### PR TITLE
Budget App : Configuration to Deployment

### PR DESCRIPTION
**Pull request by ehmaddd**
In this Pull Request, the following features are included :

### DATABASE

Database schema is designed using migration files by following this ERD :

image

---------------------------------------------------------------------------------------------------------------------------------------
### PAGES

**Splash screen**
A simple page with the name of app and links to the sign up and log in pages.

**Sign up and log in pages**
- The user must register in the app with full name, email and password (all mandatory).
- The user logs into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in.

**Home page (categories page)**
- When the user logs in, they are presented with the categories page.
- For each category, the user sees their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

**Transactions page**
- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

**"Add a new category" page**
- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

**"Add a new transaction" page**
- The user fills out a form to create a new transaction with:
  -  name (mandatory)
  -  amount (mandatory)
  -  categories (auto-selected)
- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

------------------------------------------------------------------------------------------------------------------------------------

### DESIGN

The design guidelines related to :
- Colors.
- Typography: font face, size and weight.
- Layout: composition and space between elements.

are followed according to the Original design idea by Gregoire Vella on Behance

-------------------------------------------------------------------------------------------------------------------------------------

### GENERAL

- There is no linter errors
- Used correct gitflow
- Documented work in a professional way
- Followed the list of best practices for Ruby

-------------------------------------------------------------------------------------------------------------------------------------

### INTERFACES

The following are the original interfaces of the application :

**SPLASH SCREEN**
image

**SIGNUP PAGE**
image

**SIGNIN PAGE**
image

**CATEGORIES PAGE**
image

**ADD NEW CATEGORY/GROUP**
image

**TRANSACTIONS PAGE**
image

**ADD NEW TRANSACTION**
image

-------------------------------------------------------------------------------------------------------------------------------------

### VIDEO LINK

The link for Detailed Video description is :
https://drive.google.com/file/d/1xWLNqM_F3atrUFJF3_oDLHnw9NMDFCUO/view?usp=sharing

### CODE REVIEW

File: app/views/groups/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <div class="group-element">
+      <img src="<%= group.icon %>" class="no-underline group-image">
+    </div>
+    <%= link_to group_deals_path(group), class:"group-link group-element" do %>
+      <p><%= group.name %></p>
+    <% end %>
+    <p class="group-element"><%= number_to_currency(group.deals.where(author: current_user).sum(:amount)) %></p>
+  </div>
+<% end %>
+<%= link_to 'Add a New Category', new_group_path, class:'new-category-link'%>
```
Comment (tugsem) [inline app/views/groups/index.html.erb:17]: - [ ] As per the requirement, kindly move your button to the bottom of the page and style it according to the given design. The same thing applies for the 'add a new transaction' button. Please fix it as well: 

yours|given
---|---
image|image

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @ehmaddd 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️video presented professionally
✔️app deployed properly
✔️style guidelines mostly used
✔️ app was tested thoroughly
## Required Changes ♻️
- [x] Kindly resolve these errors that appeared during testing to make your project more error-prone.
image

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/groups/index.html.erb (hunk 1 above)
Comment (ehmaddd) [reply app/views/groups/index.html.erb:17]: Removed all the errors
image

Comment (ehmaddd) [reply app/views/groups/index.html.erb:17]: Button is sticking to the bottom as required :

image

Comment (franclobo) [inline app/views/groups/index.html.erb:17]: - **[OPTIONAL]** Please 🙏🏽 take in mind that it is a mobile web app, when I checked with a width of 375px the UI should be similar to the original design, i.e. the button should cover the width. 👇🏽

image

**Review by franclobo — APPROVED** [review]
Hi @ehmaddd, 

# STATUS: APPROVED 😃 

leonardo-dicaprio-leo-dicaprio

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights
- No linter errors ✔️ 
- Use devise ✔️ 
- All tests passed ✔️ 
- Correct schema ✔️ 

Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**